src/compile/common.ts

```diff
@@ -177,3 +189,23 @@ export function titleMerger(v1: Explicit<string>, v2: Explicit<string>) {
       v1.value + ', ' + v2.value // join title with comma if different
   };
 }
+
+export function guideLabels(fieldDef: FieldDef<string>, config: Config, model: Model, channel: SpatialScaleChannel | NonspatialScaleChannel, label: any) {
+    if (fieldDef.type === TEMPORAL) {
+    const isUTCScale = model.getScaleComponent(channel).get('type') === ScaleType.UTC;
+    return {
+      signal: timeFormatExpression('datum.value', fieldDef.timeUnit, label.format, config.legend.shortTimeLabels, config.timeFormat, isUTCScale)
```

Comment:
guideLabels, but why read `config.legend`?

Just passing in `shortTimeLabels` would be less risky.